<div>
    <div class="column-sub" style="width:1%">
        &nbsp;
    </div>

    <div class="column-prime" style="width:75%">
        <br/>
        <div class="section far2x">

            <b>Welcome to the BV-BRC SARS-CoV-2 Real-time Tracking and Early Warning System for Variants and Lineages of Concern (VoCs/LoCs)</b>
            <p>
                VoCs/LoCs are sequence variants or lineages that may significantly affect vaccine efficacy, transmissibility, disease outcomes,
                or other factors critical to gaining control of COVID-19 disease <br/>
                The <b>SARS-CoV-2 Variants and Lineages of Concern</b> resource
                <ul>
                    <li>Identifies and tracks emerging variants and lineages through daily processing of publicly available SARS-CoV-2 sequences</li>
                    <li>Performs risk assessment on the variants to identify candidate VoCs/LoCs by leveraging a growing knowledgebase of
                        sequence features, including protein domains, functional regions, and immune epitopes</li>
                    <li>Provides detailed information about each VoC/LoC</li>
                    <li>Provides their sequence prevalence in various countries and regions over time using interactive dashboards and charts</li>
                    <li>Provides integrated view of the VoCs/LoCs and important sequence features using genome browsers, protein structure viewers, and phylogenetic trees</li>
                </ul>
                <br/>
                For more information, please see <a href="/patric/pdf/sars-cov-2-loc-tutorial.pdf" target=_blank>The BV-BRC Real-time Tracking and Early Warning System for SARS-CoV-2 Variants and Lineages of Concern (VoCs/LoCs) Tutorial</a>.
            </p>
        </div>
        <div class="section far2x">
            <h3 class="close2x section-title"><span class="wrap">Lineages of Concern</span></h3>
            <table class="p3basic striped">
                <thead>
                    <tr>
                        <th>LoC name</th>
                        <th>PANGO lineage</th>
                        <th>NextStrain lineage</th>
                        <th>Other synonyms</th>
                        <th>Emergence date</th>
                        <th>Emergence location</th>
                        <th>AA substitutions in spike protein</th>
                        <th>Impact</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.1.7">B.1.1.7</a></td>
                        <td><a href="https://cov-lineages.org/global_report_B.1.1.7.html" target=_blank>B.1.1.7</a></td>
                        <td>20I/501Y.V1</td>
                        <td>VOC 202012/01, variant originating in UK</td>
                        <td>September 2020</td>
                        <td>Southeast England</td>
                        <td>H69-**, V70-**, N501Y**, A570D, D614G**, P681H, T716I, S982A, D1118H</td>
                        <td>Increased transmissibility; S gene target failure (SGTF)</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.351">B.1.351</a></td>
                        <td><a href="https://cov-lineages.org/global_report_B.1.351.html" target=_blank>B.1.351</a></td>
                        <td>20H/501Y.V2</td>
                        <td>variant originating in South African</td>
                        <td>October 2020</td>
                        <td>Nelson Mandela Bay, South African</td>
                        <td>(L18F*)**, D80A, D215G, L242-, A243-, L244-, (R246I*), K417N**, E484K**, N501Y**, D614G**, A701V</td>
                        <td>E484K appears to result in loss of serum antibody neutralization; K417 is also found in RBD and may contribute to loss of serum antibody neutralization</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=P.1">P.1</a></td>
                        <td><a href="https://cov-lineages.org/lineages/lineage_B.1.1.28.html" target=_blank>B.1.1.28</a></td>
                        <td>20J/501Y.V3</td>
                        <td>variant originating in Brazil, B.1.1.248</td>
                        <td>July 2020</td>
                        <td>Brazil</td>
                        <td>L18F**, T20N, P26S, D138Y, R190S, K417T**, E484K**, N501Y**, D614G**, H655Y, T1027I, V1176F</td>
                        <td>Increased transmissibility; E484K appears to result in loss of serum antibody neutralization; K417 is also found in RBD and may contribute to loss of serum antibody neutralization</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=CAL.20C">CAL.20C</a></td>
                        <td><a href="https://cov-lineages.org/lineages/lineage_B.1.429.html" target=_blank>B.1.429</a></td>
                        <td>20C/S:452R</td>
                        <td>variant originating in California</td>
                        <td>July 2020</td>
                        <td>Southern California, USA</td>
                        <td>S13I, W152C, L452R, D614G**</td>
                        <td>Unknown - however L452R is within RBD and may result in loss of monoclonal antibody binding </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.375">B.1.375</a></td>
                        <td><a href="https://cov-lineages.org/lineages/lineage_B.1.375.html" target=_blank>B.1.375</a></td>
                        <td>&nbsp;</td>
                        <td>non-B.1.1.7 SGTF</td>
                        <td>September 2020</td>
                        <td>Massachusetts, USA</td>
                        <td>H69-**, V70-**, D614G** </td>
                        <td>S gene target failure (SGTF)</td>
                    </tr>
                </tbody>
            </table>
            * denotes mutations found in at least >25% variants, but not in all sequences</li>
            <br> ** denotes mutations found in more than one VoC indicating possible positive selection, shared ancestry and/or convergent evolution
        </div>
        <br/>
        <div class="section">
            <h3 class="close2x section-title"><span class="wrap">Variants of Concern</span></h3>
            Will be implemented soon.
        </div>
    </div>

    <div class="column-opt">
        <br/>
        <div class="section far2x">
            <h3 class="close2x section-title"><span class="wrap">Acknowledgements</span></h3>
            <p>
                We gratefully acknowledge the authors, originating and submitting laboratories that have shared their SARS-CoV-2 genomic data
                via <a href="https://www.ncbi.nlm.nih.gov/sars-cov-2/" target=_blank>GenBank and SRA</a> and <a href="https://www.cogconsortium.uk/" target=_blank>COG-UK</a>, which is used to build this system.
            </p>
        </div>
        <div class="section">
            <h3 class="close2x section-title"><span class="wrap">Recent PubMed Articles</span></h3>
            <div data-dojo-attach-point="pubmedSummaryNode">
                Loading...
            </div>
        </div>
    </div>
</div>
