<div>
    <div class="column-sub" style="width:1%">
        &nbsp;
    </div>

    <div class="column-prime" style="width:75%">
        <br/>
        <div class="section far2x">

            <b>Welcome to the BV-BRC SARS-CoV-2 Real-time Tracking and Early Warning System for Variants and Lineages of Concern (VoCs/LoCs)</b>
            <p>
                VoCs/LoCs are sequence variants or lineages that may significantly affect vaccine efficacy, transmissibility, disease outcomes,
                or other factors critical to gaining control of COVID-19 disease <br/>
                The <b>SARS-CoV-2 Variants and Lineages of Concern</b> resource
                <ul>
                    <li>Identifies and tracks emerging variants and lineages through daily processing of publicly available SARS-CoV-2 sequences</li>
                    <li>Performs risk assessment on the variants to identify candidate VoCs/LoCs by leveraging a growing knowledgebase of
                        sequence features, including protein domains, functional regions, and immune epitopes</li>
                    <li>Provides detailed information about each VoC/LoC</li>
                    <li>Provides their sequence prevalence in various countries and regions over time using interactive dashboards and charts</li>
                    <li>Provides integrated view of the VoCs/LoCs and important sequence features using genome browsers, protein structure viewers, and phylogenetic trees</li>
                </ul>
                <br/>
                For more information, please see <a href="/patric/pdf/sars-cov-2-loc-tutorial.pdf" target=_blank>The BV-BRC Real-time Tracking and Early Warning System for SARS-CoV-2 Variants and Lineages of Concern (VoCs/LoCs) Tutorial</a>.
            </p>
        </div>
        <div class="section close2x">
            <h3 class="close2x section-title"><span class="wrap">Lineages of Concern</span></h3>
            <table class="p3basic striped">
                <thead>
                    <tr>
                        <th>LoC name</th>
                        <th>PANGO lineage</th>
                        <th>NextStrain lineage</th>
                        <th>Other synonyms</th>
                        <th>Emergence date</th>
                        <th>Emergence location</th>
                        <th>AA substitutions in spike protein</th>
                        <th>Impact</th>
                    </tr>
                </thead>
                <tbody>
                    <tr><!-- B.1.1.7 -->
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.1.7">B.1.1.7</a></td>
                        <td><a href="https://cov-lineages.org/global_report_B.1.1.7.html" target=_blank>B.1.1.7</a></td>
                        <td>20I/501Y.V1</td>
                        <td>VOC 202012/01, variant originating in UK</td>
                        <td>September 2020</td>
                        <td>Southeast England</td>
                        <td>H69-**, V70-**, N501Y**, A570D, D614G**, P681H, T716I, S982A, D1118H</td>
                        <td>Increased transmissibility; S gene target failure (SGTF);
                            <a href="https://www.nature.com/articles/s41586-021-03426-1)" target=_blank>increased severity</a>;
                            <a href="https://www.fda.gov/media/145802/download" target=_blank>little impact on neutralization by moclonal antibodies</a>;
                            litte impact on neautralization by polyclonal antibodies
                            <a href="https://doi.org/10.1101/2021.01.25.428137" target=_blank>[1]</a>
                            <a href="https://doi.org/10.1101/2021.01.27.428516" target=_blank>[2]</a>
                        </td>
                    </tr>
                    <tr><!-- B.1.351 -->
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.351">B.1.351</a></td>
                        <td><a href="https://cov-lineages.org/global_report_B.1.351.html" target=_blank>B.1.351</a></td>
                        <td>20H/501Y.V2</td>
                        <td>variant originating in South African</td>
                        <td>October 2020</td>
                        <td>Nelson Mandela Bay, South African</td>
                        <td>(L18F*)**, D80A, D215G, L242-, A243-, L244-, (R246I*), K417N**, E484K**, N501Y**, D614G**, A701V</td>
                        <td>
                            Increased transmissibility (Pearson CAB, Russell TW, Davies NG, et al. Estimates of severity and transmissibility of novel South Africa SARS-CoV-2 variant 501Y.V2);
                            E484K appears to result in loss of serum antibody neutralization;
                            K417 is also found in RBD and may contribute to loss of serum antibody neutralization
                        </td>
                    </tr>
                    <tr><!-- B.1.427 -->
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.427">B.1.427</a></td>
                        <td><a href="https://cov-lineages.org/lineages/lineage_B.1.427.html" target=_blank>B.1.427</a></td>
                        <td>20C/S:452R</td>
                        <td>CAL.20C</td>
                        <td>September 2020</td>
                        <td>Southern California, USA</td>
                        <td>S13I, W152C,  L452R, D614G**</td>
                        <td>
                            <a href="https://doi.org/10.1101/2021.03.07.21252647" target=_blank>increased transmissibility</a>;
                            deacreased neutralization by some monoclonal antibodies;
                            <a href="https://doi.org/10.1101/2021.03.07.21252647" target=_blank>decreased neutralization by polyclonal anibodies </a>
                        </td>
                    </tr>
                    <tr><!-- B.1.429 -->
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.429">B.1.429</a></td>
                        <td><a href="https://cov-lineages.org/lineages/lineage_B.1.429.html" target=_blank>B.1.429</a></td>
                        <td>20C/S:452R</td>
                        <td>CAL.20C</td>
                        <td>July 2020</td>
                        <td>Southern California, USA</td>
                        <td>S13I, W152C, L452R, D614G**</td>
                        <td>
                            <a href="https://doi.org/10.1101/2021.03.07.21252647" target=_blank>increased transmissibility</a>;
                            deacreased neutralization by some monoclonal antibodies;
                            <a href="https://doi.org/10.1101/2021.03.07.21252647" target=_blank>decreased neutralization by polyclonal anibodies </a>
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=P.1">P.1</a></td>
                        <td><a href="https://cov-lineages.org/lineages/lineage_B.1.1.28.html" target=_blank>B.1.1.28.1</a></td>
                        <td>20J/501Y.V3</td>
                        <td>variant originating in Brazil, B.1.1.248</td>
                        <td>July 2020</td>
                        <td>Brazil</td>
                        <td>L18F**, T20N, P26S, D138Y, R190S, K417T**, E484K**, N501Y**, D614G**, H655Y, T1027I, V1176F</td>
                        <td>Increased transmissibility; E484K appears to result in loss of serum antibody neutralization; K417 is also found in RBD and may contribute to loss of serum antibody neutralization</td>
                    </tr>
                </tbody>
            </table>
            * denotes mutations found in at least >25% variants, but not in all sequences</li>
            <br> ** denotes mutations found in more than one VoC indicating possible positive selection, shared ancestry and/or convergent evolution
        </div>
        <br/>
        <div class="section far2x">
            <h3 class="close2x section-title"><span class="wrap">Lineages of Interest</span></h3>
            <table class="p3basic striped">
                <thead>
                    <tr>
                        <th>LoI name</th>
                        <th>PANGO lineage</th>
                        <th>NextStrain lineage</th>
                        <th>Other synonyms</th>
                        <th>Emergence date</th>
                        <th>Emergence location</th>
                        <th>AA substitutions in spike protein</th>
                        <th>Impact</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=A.23.1">A.23.1</a></td>
                        <td>A.23.1 + E484K</td>
                        <td>-</td>
                        <td>VUI 202102/01</td>
                        <td>2020-10-21</td>
                        <td>Uganda</td>
                        <td>R102I, F157L, V367F, E484K, Q613H, P681R</td>
                        <td>
                            Q613H is predicted to be functionally equivalent to the D614G mutation that arose early in 2020.
                            International lineage with a number of variants of potential biological concern, including 681R
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=A.27">A.27</td>
                        <td>A.27</td>
                        <td>19B/501Y</td>
                        <td>-</td>
                        <td>2020-12-14</td>
                        <td>France</td>
                        <td>L18F, L452R, N501Y, D614D, A653V, H655Y, D796Y, G1219V (sometimes Q677H)</td>
                        <td>
                            Mayotte/ European cluster with 18 changes in the long branch leading to this cluster,
                            including 7 NS changes in the spike (L18F, L452R, N501Y, A653V, H655Y, D796Y, G1219V), on a 614D background.
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.1.318">B.1.1.318 + E484K</a></td>
                        <td>B.1.1.318</td>
                        <td>-</td>
                        <td>VUI 202102/04</td>
                        <td>2021-02</td>
                        <td>England</td>
                        <td>T95I, del144*, E484K, P681H, D796H</td>
                        <td>
                            Lineage circulating in multiple countries with a number of variants of concern
                            (Spike D614G, Spike D796H, Spike E484K, Spike P681H, Spike T95I, Spike Y144del).
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.1.519">B.1.1.519</a></td>
                        <td>B.1.1.519</td>
                        <td>-</td>
                        <td>-</td>
                        <td>2020-10</td>
                        <td>Mexico</td>
                        <td>T478K, D614G, P681H, T732A </td>
                        <td>USA/ Mexico lineage</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.525">B.1.525</a></td>
                        <td>B.1.525</td>
                        <td>20A/S:484K</td>
                        <td>-</td>
                        <td>2020-12-15</td>
                        <td>Multiple</td>
                        <td>Q52R, E484K, D614G, Q677H, F888L</td>
                        <td>
                            Potential decreased neutralization by monoclonal and polyclonal antibodies.
                            International lineage with E484K, del69-70 among other defining mutations.
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.526">B.1.526</a></td>
                        <td>-</td>
                        <td>20C/S:484K</td>
                        <td>NY variant</td>
                        <td>2020-11-15</td>
                        <td>New York</td>
                        <td>L5F, T95I, D253G, E484K*, D614G, A701V</td>
                        <td>
                            Minimal decrease in neutralization by monoclonal and polyclonal antibodies
                            <a href="https://www.biorxiv.org/content/10.1101/2021.03.24.436620v1" target=_blank>[1]</a>
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.526.1">B.1.526.1</a></td>
                        <td>-</td>
                        <td>20C</td>
                        <td>-</td>
                        <td>2020-12</td>
                        <td>New York</td>
                        <td>D80G, Y144-, F157S, L452R, D614G, T859N, D950H</td>
                        <td>
                            Potential decrease in neutralization by monoclonal and polyclonal antibodies.
                            Sublineage of B.1.526 (with spike mutations T95I and D253G) that appears to have several more unique mutations.
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.526.2">B.1.526.2</a></td>
                        <td>-</td>
                        <td>20C</td>
                        <td>-</td>
                        <td>2020-12-08</td>
                        <td>New York</td>
                        <td>L5F, T95I, D253G, S477N, D614G, Q957R</td>
                        <td>
                            Potential decrease in neutralization by monoclonal and polyclonal antibodies.
                            Sublineage of B.1.526 (with spike mutations T95I and D253G) that appears to have several more unique mutations.
                        </td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.617.1">B.1.617.1</a></td>
                        <td>B.1.617.1</td>
                        <td>-</td>
                        <td>B.1.617</td>
                        <td>2020-12-01</td>
                        <td>India</td>
                        <td>E154K, L452R, E484Q, D614G, P681R, Q1071H</td>
                        <td>Predominantly India lineage with 484Q</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.617.2">B.1.617.2</a></td>
                        <td>B.1.617.2</td>
                        <td>-</td>
                        <td>B.1.617</td>
                        <td>2020-12-12</td>
                        <td>UK</td>
                        <td>T19R, L452R, T478K, D614G, P681R,  D950N</td>
                        <td>Predominantly India lineage with several spike mutations</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=B.1.617.3">B.1.617.3</a></td>
                        <td>B.1.617.3</td>
                        <td>-</td>
                        <td>B.1.617</td>
                        <td>2020-10-02</td>
                        <td>India</td>
                        <td>T19R, L452R, E484Q, D614G, P681R</td>
                        <td>Predominantly India lineage with 484Q</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=C.37">C.37</a></td>
                        <td>C.37</td>
                        <td>20J</td>
                        <td>B.1.1.1.37</td>
                        <td>2021-01</td>
                        <td>Peru</td>
                        <td>G75V, T76I, R246N, S247-, Y248-, L249-, T250-, P251-, G252-, D253-, L452Q, F490S, D614G, T859N</td>
                        <td>Several deletions in spike protein- 2 escape mutations- sublineage may have S gene dropout</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=P.2">P.2</a></td>
                        <td>P.2</td>
                        <td>-</td>
                        <td>B.1.1.28.2</td>
                        <td>2020-04-15</td>
                        <td>Brazil</td>
                        <td>E484K, D614G, V1176F</td>
                        <td>Potential decreased neutralization by monoclonal and polyclonal antibodies</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=P.3">P.3</a></td>
                        <td>P.3</td>
                        <td>-</td>
                        <td>B.1.1.28.3; VUI-21MAR-02</td>
                        <td>2021-01-16</td>
                        <td>Philippines</td>
                        <td>L141-, G142-, V143-, A243-, L244-, E484K, N501Y, D614G, P681H, E1092K, H1101Y, V1176F</td>
                        <td>Potential decrease in neutralization by monoclonal and polyclonal antibodies. Alias of B.1.1.28.3, Lineage predominantly in the Philippines with spike mutations E484K, N501Y, P681H, 141-143del</td>
                    </tr>
                    <tr>
                        <td><a href="/view/VariantLineage/#view_tab=lineage&loc=R.1">R.1</a></td>
                        <td>R.1</td>
                        <td>-</td>
                        <td>B.1.1.316.1</td>
                        <td>2020-10-24</td>
                        <td>Multiple</td>
                        <td>W152L, E484K, D614G, G769V</td>
                        <td>Alias of B.1.1.316.1, Sublineage of B.1.1.316 with 3 additional spike mutations, circulating in a number of countries.</td>
                    </tr>
                </tbody>
            </table>
            * denotes mutations found in at least >25% variants, but not in all sequences</li>
            <br> ** denotes mutations found in more than one VoC indicating possible positive selection, shared ancestry and/or convergent evolution
        </div>
    </div>

    <div class="column-opt">
        <br/>
        <div class="section far2x">
            <h3 class="close2x section-title"><span class="wrap">Acknowledgements</span></h3>
            <p>
                We gratefully acknowledge the authors, originating and submitting laboratories that have shared their SARS-CoV-2 genomic data
                via <a href="https://www.ncbi.nlm.nih.gov/sars-cov-2/" target=_blank>GenBank and SRA</a> and <a href="https://www.cogconsortium.uk/" target=_blank>COG-UK</a>, which is used to build this system.
            </p>
        </div>
        <div class="section">
            <h3 class="close2x section-title"><span class="wrap">Recent PubMed Articles</span></h3>
            <div data-dojo-attach-point="pubmedSummaryNode">
                Loading...
            </div>
        </div>
    </div>
</div>
