<div>
    <div style="width: 78%">
        &nbsp;
        <blockquote style="padding: 5px; float:right">
            Last updated on 04/07/2025
        </blockquote>
    </div>
    <br/>
    <div class="column-sub" style="width:15%" data-dojo-attach-point="leftPanelHTML">
        <div class="section" style="padding-left: 10px;">
            <h3 class="close2x section-title" style="margin-top: 0;"><span class="wrap">Contents</span></h3>
            <div class="pubmed">
                <ul>
                    <li><a href="#search-genome-section">Search Genome</a></li>
                    <li><a href="#download-annotation-section">Download Annotation</a>
                        <ul style="margin-top: 5px;border-top: #aec9d2 1px solid;padding: 5px 0 0 7px;">
                            <li><a href="#download-genometype-section">Download the GenomeType</a></li>
                            <li style="border-bottom: none; margin: 0; padding-bottom: 0;"><a
                                    href="#convert-genometype-section">Convert the GenomeType to Desired format</a></li>
                        </ul>
                    </li>
                    <li><a href="#download-genome-data-section">Download genome data in FASTA format</a></li>
                </ul>
            </div>
        </div>
    </div>

    <div class="column-prime" style="width:63%">
        <div class="section far2x">
            <p style="margin-bottom: 5px;">
                Below are some of useful command line tool reference and examples to search and download bulk data from
                BV-BRC. Complete command line interface (CLI) guide is available at <a
                    href="https://www.bv-brc.org/docs/cli_tutorial/" target="_blank">Command Line Interface</a>. CLI
                installation guide is available at <a
                    href="https://www.bv-brc.org/docs/cli_tutorial/cli_installation.html#installing-the-bv-brc-command-line-interface"
                    target="_blank">Installing the BV-BRC Command Line Interface</a>.
            </p>
        </div>
        <div class="section far2x" id="search-genome-section">
            <h3 class="close2x section-title"><span class="wrap">Search Genome</span></h3>
            <p style="margin-bottom: 5px;">
                User can search genomes using <code style="font-size: 100%"><b>p3-all-genomes</b></code> command.
                Complete help and guide are at <a
                    href="https://www.bv-brc.org/docs/cli_tutorial/command_list/p3-all-genomes.html"
                    target="_blank">p3-all-genomes</a>.
            </p>
            <p style="margin-bottom: 5px;">
                Example of p3-all-genomes command.
            </p>
            <p style="margin-bottom: 5px;">
                Following command is to download measles genomes collected in 2025. Please make sure you format the date
                correctly.
            </p>
            <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">$ p3-all-genomes --eq taxon_lineage_ids,11234 --eq collection_year,2025 --in genome_status,Complete,Partial
genome.genome_id
658048.5627
658048.5626
658048.5624
658048.5625
658048.5623
170528.8105
658048.5738</code></pre>
            <p style="margin-bottom: 5px;">
                Following commands will get the genome id of all measles genomes from 2024 from USA in FASTA
                format.
            </p>
            <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">$ p3-all-genomes \
  --eq=taxon_lineage_ids,11234 \
  --eq=isolation_country,USA \
  --eq=host_common_name,* \
  --eq=collection_year,2024 \
  | head -n 10 > measles_virus.txt

<b>#Example output</b>
$ more measles_virus.txt
genome.genome_id
170528.6574
170528.6572
170528.6571
</code></pre>
        </div>
        <div class="section far2x" id="download-annotation-section">
            <h3 class="close2x section-title"><span class="wrap">Download Annotation</span></h3>
            <p style="margin-bottom: 5px;">
                User can download annotation in two steps.
            </p>
            <ol style="margin-bottom: 5px;">
                <li>Download the GenomeType</li>
                <li>Convert the GenomeType to desired format.</li>
            </ol>
            <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">GenomeType conversion format options
genbank         Genbank format
genbank_merged  Genbank format as single merged locus.
feature_data    Tabular form of feature data
protein_fasta   Protein translations in fasta format
contig_fasta    Contig DNA in fasta format
feature_dna     Feature DNA sequences in fasta format
gff             GFF format
embl            EMBL format
</code></pre>
            <blockquote style="margin-bottom: 5px;">
                <b>More details of commands available at <a
                        href="https://www.bv-brc.org/docs/cli_tutorial/command_list/p3-gto.html"
                        target="_blank">p3-gto</a> and <a
                        href="https://www.bv-brc.org/docs/cli_tutorial/rasttk_incremental_commands.html"
                        target="_blank">RASTtk, The Incremental Commands</a>.</b>
            </blockquote>
            <section id="download-genometype-section">
                <h3 style="margin-top: 12px;margin-bottom: 7px;"><span style="font-size: 15px;line-height: 20px;">1. Download the GenomeType</span>
                </h3>
                <p style="margin-bottom: 5px;">
                    First, download the GenomeType using the Genome ID.
                </p>
                <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">p3-gto [Genome ID]</code></pre>
                <p style="margin-bottom: 5px;">
                    Example:
                </p>
                <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">p3-gto 170528.6574</code></pre>
            </section>
            <section id="convert-genometype-section">
                <h3 style="margin-top: 12px;margin-bottom: 7px;"><span style="font-size: 15px;line-height: 20px;">2. Convert GenomeType to Desired Format</span>
                </h3>
                <p style="margin-bottom: 5px;">
                    Once you have the GenomeType file, you can convert it to your desired format.
                </p>
                <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">rast-export-genome [Format] < [GenomeType File] > [Output File]</code></pre>
                <ul>
                    <li>Replace <b>[Format]</b> with one of the conversion options (e.g. gff, genbank, protein_fasta,
                        etc.).
                    </li>
                    <li>Replace <b>[GenomeType File]</b> with the filename of the downloaded GenomeType file.</li>
                    <li>Replace <b>[Output File]</b> with the desired name for the converted file.</li>
                </ul>
                <p style="margin-bottom: 5px;">
                    Example:
                </p>
                <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">rast-export-genome gff < 170528.6574.gto > 170528.6574.gto.gff</code></pre>
            </section>
        </div>
        <div class="section far2x" id="download-genome-data-section">
            <h3 class="close2x section-title"><span class="wrap">Download genome data in FASTA format</span></h3>
            <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">$ p3-genome-fasta -h
p3-genome-fasta.pl [-h] [long options...] genomeid
--protein  feature protein fasta
--feature  feature dna fasta
--contig   contig dna fasta
-h --help  display usage information
</code></pre>
            <p style="margin-bottom: 5px;">
                Example of downloading genome using genome id:
            </p>
            <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">$ p3-genome-fasta --contig 170528.6574
>170528.6574.con.0001
gtcagttccacattggcatctgaactcggtatcactgccgaggatgcaaggcttgtttca
gagattgcaatgcacaccaccgaggacaggaccagcagagcagttggacccagacaagct
â€¦
</code></pre>
            <p style="margin-bottom: 5px;">
                Example of searching and downloading data using pipe command:
            </p>
            <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code">p3-all-genomes  --eq taxon_lineage_ids,11234 --gt date_inserted,"2024-08-01T00:00:00.000Z" \
 | sed -n '2,$p' | xargs -n 1 p3-genome-fasta --contig</code></pre>
            <p style="margin-bottom: 5px;">
                Search genome list and then download data based on genome list:
            </p>
            <pre class="bv-brc-code-block" tabindex="0"><code class="bv-brc-code"><b>#Search the genome list</b>
$ p3-all-genomes  --eq taxon_lineage_ids,11234 --gt date_inserted,"2024-08-01T00:00:00.000Z" \
 > genome_list.txt

<b>#Download fasta file for genome list. You can remove the genomes for genome-list as needed</b>
$ for i in $(cat genome_list.txt); do
    if [ $i != "genome.genome_id" ]; then
       p3-genome-fasta --contig $i
    fi
 done >> genome.fasta

<b>#View the genome fasta file</b>
$ more genome.fasta
>170528.7065.con.0001
atggccacacttttaaggagcttagcattgttcaaaagaaacaaggacaaaccacccatt
acatcaggatccggtggagccatcagaggaatcaaacacattattatagtaccaatccct
ggagattcctcaattaccacccgatccaggctactggaccgattggtcaggttaattgga
aacccagatgtgagcggacccaaactaacaggggcactaataggtatattatctttgttt
...</code></pre>
        </div>
    </div>
</div>