<div>
    <br>
    <div class="column-sub"></div>

    <div class="column-prime" style="width:50%">
        <div class="section close2x" id="data-section" style="padding-left: 10px;">
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/view/Taxonomy/11234#view_tab=genomes&filter=or(eq(collection_year,"2025"),eq(collection_year,"2026"))&defaultColumns=-cds,subclade,collection_date&defaultSort=genome_name,subclade'
                   target="_blank">Measles Outbreak Genomes <i class="icon-external-link"></i></a>
            </h4>
            <p style="margin-bottom: 15px;">
                Measles genomes from 2025 and 2026
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/view/Taxonomy/11234#view_tab=genomes&defaultColumns=-cds,subclade,collection_date&defaultSort=-collection_year,genome_name,subclade'
                   target="_blank">Measles Outbreak Genomes <i class="icon-external-link"></i></a>
            </h4>
            <p style="margin-bottom: 15px;">
                All Measles genomes
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href="/workspace/mkuscuog@bvbrc/Outbreaks/Measles2025/files" target="_blank">Measles outbreak FASTA,
                    MSA and phylogenetic tree files from 2025 available in the BV-BRC workspace <i
                            class="icon-external-link"></i></a>.
            </h4>
            <p style="margin-bottom: 5px;">
                These files contain the data used to generate the phylogenetic trees provided on the Phylogenetics tab
                on the Measles Outbreak page.
            </p>
            <p style="margin-bottom: 5px;">
                Select genome sequences were downloaded from NCBI and BV-BRC.
            </p>
            <p style="margin-bottom: 5px;">
                Sequences were then aligned using MAFFT. Multiple sequence alignment columns with more than 50% gaps
                were deleted. A minimal evolution phylogenetic tree was then inferred with the FastME program based on
                maximum likelihood pairwise distances calculated by TREE-PUZZLE using the GTR model. Custom scripts were
                used to add meta-data to the resulting tree.
            </p>
        </div>
    </div>
</div>
