<div>
    <br>
    <div class="column-sub"></div>

    <div class="column-prime" style="width:50%">
        <div class="section close2x" id="data-section" style="padding-left: 10px;">
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/view/Taxonomy/10244#view_tab=genomes&filter=or(eq(collection_year,"2024"),eq(collection_year,"2025"))&defaultColumns=-cds,clade,collection_date&defaultSort=genome_name,clade' target="_blank">Mpox
                    Outbreak Genomes <i class="icon-external-link"></i></a>
            </h4>
            <p style="margin-bottom: 15px;">
                All Mpox genomes from 2024 and 2025
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/searches/SFVTSearch?pathogen_group=Monkeypox%20virus' target="_blank">Sequence Feature Variant Type (SFVT) <i class="icon-external-link"></i></a>
            </h4>
            <p style="margin-bottom: 15px;">
                Search for Sequence Feature Variant Type data
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href="ftp://ftp.bvbrc.org/outbreaks/Mpox_2024" target="_blank">Monkeypox outbreak FASTA,
                    MSA and phylogenetic tree files from 2024 and 2025 available from the BV-BRC FTP Site <i
                            class="icon-external-link"></i></a>.
            </h4>
            <p style="margin-bottom: 5px;">
                These files contain the data used to generate the phylogenetic trees provided on the Phylogenetics tab
                on the MPOX Outbreak page.
            </p>
            <p style="margin-bottom: 5px;">
                Select genome sequences were downloaded from NCBI and BV-BRC to cover all major clades.
            </p>
            <p style="margin-bottom: 5px;">
                Sequences were then aligned using MAFFT. Multiple sequence alignment columns with more than 50% gaps
                were deleted. A minimal evolution phylogenetic tree was then inferred with the FastME program based on
                maximum likelihood pairwise distances calculated by TREE-PUZZLE using the GTR model and 100 bootstrap
                re-samplings. Custom scripts were used to add meta-data to the resulting tree. Clade information is
                based on NEXSTRAIN monkeypox virus phylogenetics.
            </p>
        </div>
    </div>
</div>
