<div>
    <br>
    <div class="column-sub"></div>

    <div class="column-prime" style="width:50%">
        <div class="section close2x" id="data-section" style="padding-left: 10px;">
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/view/Taxonomy/11320#view_tab=genomes&filter=and(eq(subtype,"H5N1"),or(eq(collection_year,"2024"),eq(collection_year,"2025")),or(eq(genome_status,"Complete"),eq(genome_status,"Partial")))&defaultColumns=-cds,h5_clade,subclade,segment,sra_accession,collection_date&defaultSort=genome_name,segment'>H5N1
                    Outbreak Genomes</a>
            </h4>
            <p style="margin-bottom: 15px;">
                All H5N1 genomes from 2024 and 2025
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/view/Taxonomy/11320#view_tab=genomes&filter=and(eq(subtype,"H5N1"),or(eq(collection_year,"2024"),eq(collection_year,"2025")),eq(genbank_accessions,*),or(eq(genome_status,"Complete"),eq(genome_status,"Partial")))&defaultColumns=-cds,h5_clade,subclade,segment,sra_accession,collection_date&defaultSort=genome_name,segment'>H5N1
                    Outbreak genomes from GenBank</a>
            </h4>
            <p style="margin-bottom: 15px;">
                H5N1 genomes from 2024 and 2025 that have GenBank accession numbers.
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/view/Taxonomy/11320#view_tab=genomes&filter=and(eq(subtype,"H5N1"),or(eq(collection_year,"2024"),eq(collection_year,"2025")),eq(geographic_group,"North%20America"),ne(genbank_accessions,*),or(eq(genome_status,"Complete"),eq(genome_status,"Partial")))&defaultColumns=-cds,h5_clade,subclade,segment,sra_accession,collection_date,-genbank_accessions&defaultSort=genome_name,segment'>H5N1
                    Outbreak genomes assembled from SRA</a>
            </h4>
            <p style="margin-bottom: 15px;">
                H5N1 genomes from 2024 and 2025 in North America assembled from the NCBI Short Read Archive (SRA) database. The
                BV-BRC assembled and annotated these genomes from SRA records that are not yet linked to a GenBank
                record.
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href='/searches/SFVTSearch?pathogen_group=Influenza%20A%20virus'>Sequence Feature Variant Type (SFVT)</a>
            </h4>
            <p style="margin-bottom: 15px;">
                Search for Sequence Feature Variant Type data
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href="ftp://ftp.bvbrc.org/outbreaks/Influenza_A_virus_H5N1_2024" target="_blank">H5N1 outbreak FASTA,
                    MSA and phylogenetic tree files from 2024 and 2025 available from the BV-BRC FTP Site <i
                            class="icon-external-link"></i></a>.
            </h4>
            <p style="margin-bottom: 5px;">
                These files contain the data used to generate the phylogenetic trees provided on the Phylogenetics tab
                on the H5N1 Outbreak page.
            </p>
            <p style="margin-bottom: 5px;">
                The data is cleaned following a number of steps. For sequences prior to 2024, H5N1 DNA sequences for
                segments were downloaded from BV-BRC only from isolates for which all 8 segments were present in the
                database. This was followed by quality control (QC) and representative sequence selection:
            </p>
            <ul style="margin-bottom: 5px;">
                <li>For Segment 4, a minimum length cutoff of 1600nt was used, allowing no more than 0.1% undetermined
                    residues (“N”).
                </li>
                <li>Representative sequence selection was performed by clustering using cd-hit with a cutoff of 0.98%.
                </li>
                <li>Other segments were retained only for those isolates for which Segment 4 passed the QC and
                    clustering step.
                </li>
                <li>Sequences prior to 2003 were excluded from the analysis to prevent the resulting trees from being
                    overly long.
                </li>
            </ul>
            <p style="margin-bottom: 15px;">
                For 2024 sequences, all H5N1 sequences from BV-BRC were used without additional quality control or
                clustering. While all sequences undergo quality control, users are encouraged to review alignments
                manually before submitting them to programs specific to ancestral alignment to ensure data accuracy and
                reliability.
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1102327">H5N1 USDA Outbreak BioProject <i
                        class="icon-external-link"></i></a>
            </h4>
            <p style="margin-bottom: 15px;">
                The intent of this BioProject is to rapidly share sequences to ensure public transparency and represent
                affected cattle herds, poultry premises, peridomestic animals, and wild animals that support the
                phylogeny. Due to the urgency of sequence sharing, epidemiological investigations may still be ongoing;
                thus, sequences are initially made publicly available with limited metadata.
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href="https://www.ncbi.nlm.nih.gov/sra?term=(H5N1)%20AND%20(%222024%2F01%2F01%22%5BPublication%20Date%5D%20%3A%20%223000%22%5BPublication%20Date%5D)"
                   target="_blank">H5N1 outbreak raw reads on NCBI SRA <i class="icon-external-link"></i></a>.
            </h4>
            <p style="margin-bottom: 15px;">
                The NCBI Sequence Read Archive (SRA) is a public repository maintained by the National Center for
                Biotechnology Information (NCBI) that stores raw sequencing data. This link provides all SRA records
                with H5N1 keyword search with data published since 2024.
            </p>
            <h4 style="margin-bottom: 5px;font-weight: bold;">
                <a href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?SeqType_s=Nucleotide&VirusLineage_ss=Influenza%20A%20virus,%20taxid:11320&CollectionDate_dr=2024-01-01T00:00:00.00Z%20TO%202030-01-01T23:59:59.00Z&Serotype_s=H5N1"
                   target="_blank">H5N1 outbreak data at NCBI Virus Data <i class="icon-external-link"></i></a>.
            </h4>
            <p style="margin-bottom: 15px;">
                NCBI Virus is a specialized resource within the National Center for Biotechnology Information (NCBI)
                that focuses on providing comprehensive access to viral sequence data and related information. This
                particular link dedicated to H5N1 data since 2024.
            </p>
        </div>
    </div>
</div>
